### DISCUSSION TITLE
Nginx: get a valid certificate for the AIO interface

### DISCUSSION THREAD

**Discussion by JeCheeseSmith** [nextcloud/all-in-one#2801] [Questions] [answered]
Hi! I used the information supplied for the Reverse Proxy Setup with Nginx. (https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md )

I would still like to get a valid certificate for my AIO Interface, however I don't know that much about Webserver Configs. I'm a newbie and I really love this project.

Is there a way to "convert" this code for Caddy to a Nginx server block correctly? 

"https://<your-nc-domain>:8443 {
    reverse_proxy https://localhost:8080 {
        transport http {
            tls_insecure_skip_verify
        }
    }
}
"

**Comment by Zoey2936** [comment, marked as answer]
you can use the same config as for nextcloud and simply change the domain and port to proxy to

**Reply by JeCheeseSmith** [reply]
It costed me some time to make it work but I remarked that I should've used proxy_pass http**S**://127.0.0.1:<favored port>$request_uri; instead of HTTP. Thanks for your reply!

**Reply by dialloamadou** [reply]
how to did you fix this, I have Nginx proxy manager

**Reply by JeCheeseSmith** [reply]
Hi @dialloamadou , I use the Ubuntu Nginx Webserver Package, I don't use Nginx proxy manager. I'm a newbee to any webserver management, so I wouldn't know how to configure it using the GUI of Nginx Proxy Manager. If you read https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md , as supplied by the team. You should get a start.
